src/internal/modules/charge-information/controller.js

```diff
@@ -105,47 +107,106 @@ const postUseAbstractionData = createPostHandler(
   request => routing.getCheckData(request.pre.licence)
 );
 
-const getCheckData = async (request, h) => {
-  const { licence, draftChargeInformation } = request.pre;
+const getInvoiceAccountAddressFromDraft = draftChargeInformation => {
+  const invoiceAccountAddresses = get(draftChargeInformation, 'billingAccount.billingAccount.invoiceAccountAddresses', []);
+  const invoiceAccountAddress = get(draftChargeInformation, 'billingAccount.invoiceAccountAddress');
 
-  const invoiceAccountAddress = draftChargeInformation.billingAccount.billingAccount.invoiceAccountAddresses.find(add => {
-    return add.id === draftChargeInformation.billingAccount.invoiceAccountAddress;
-  });
+  return invoiceAccountAddresses.find(add => add.id === invoiceAccountAddress);
+};
+
+const getCheckData = async (request, h) => {
+  const { licence, draftChargeInformation, isChargeable } = request.pre;
+  const back = isChargeable
+    ? routing.getUseAbstractionData(licence)
+    : routing.getEffectiveDate(licence);
 
   const view = {
     ...request.view,
     caption: `Licence ${licence.licenceNumber}`,
-    back: routing.getUseAbstractionData(licence),
+    back,
     pageTitle: 'Check charge information',
     draftChargeInformation,
-    invoiceAccountAddress
+    licence,
+    invoiceAccountAddress: getInvoiceAccountAddressFromDraft(draftChargeInformation),
+    isChargeable
   };
 
   return h.view('nunjucks/charge-information/check.njk', view);
 };
 
 const postCheckData = async (request, h) => {
   const { nextStep } = request.payload;
-  const { licence } = request.pre;
+  const { licence, isChargeable } = request.pre;
 
   if (nextStep === 'confirm') {
     // send the draft data to the water service to save
     console.log('Collected data', request.pre.draftChargeInformation);
```

Comment:
It looks from this as though the data is being collected in a Redis key not using request.yar.

I had started building this a while ago before we decided to use a PG table in the water service for persistence of the workflow data, and this provided a way for it to be shared amongst internal users.

So there's no need to do this anymore I don't think.  All data in the flow can be stored in the user's session as normal with request.yar.